libs/designsystem/shared/floating/src/floating.directive.ts

```diff
@@ -0,0 +1,368 @@
+  }
+
+  private updateHostElementPosition(): void {
+    if (!this.reference) {
+      return;
+    }
+
+    const config: ComputePositionConfig = {
+      placement: this.placement,
+      middleware: this.getMiddlewareConfig(),
+      strategy: this.strategy,
+    } as ComputePositionConfig;
+
+    from(
+      computePosition(this.reference?.nativeElement, this.elementRef.nativeElement, config)
+    ).subscribe(({ x, y }) => this.setPositionStylingOnHostElement(x, y));
+  }
+
+  private getMiddlewareConfig(): Array<Middleware | null | undefined | false> {
+    const middleware: Array<Middleware | null | undefined | false> = [];
+    middleware.push(offset(this.offset));
+
+    if (this.shift) {
+      middleware.push(shift());
+    }
+
+    if (this.autoPlacement) {
+      middleware.push(autoPlacement());
+    }
+
+    return middleware;
+  }
+
+  private autoUpdatePosition(): void {
+    this.removeAutoUpdaterRef();
+
+    if (!this.reference) {
+      return;
+    }
+
+    this.autoUpdaterRef = autoUpdate(
+      this.reference?.nativeElement,
+      this.elementRef.nativeElement,
+      this.updateHostElementPosition.bind(this)
+    );
+  }
+
+  private setPositionStylingOnHostElement(xPosition: number, yPosition: number): void {
+    this.renderer.setStyle(this.elementRef.nativeElement, 'left', `${xPosition}px`);
+    this.renderer.setStyle(this.elementRef.nativeElement, 'top', `${yPosition}px`);
+    this.renderer.setStyle(
+      this.elementRef.nativeElement,
+      'display',
+      this.isShown ? `block` : `none`
+    );
+  }
+
+  private setupEventHandling(): void {
+    if (!this.reference || !this.triggers) {
+      return;
+    }
+
+    this.triggers.forEach((trigger: TriggerEvent) =>
+      this.attachTriggerEventToReferenceElement(trigger)
+    );
+  }
+
+  private attachTriggerEventToReferenceElement(trigger: TriggerEvent): void {
+    const events: EventMethods[] | undefined = this.triggerEventMap.get(trigger);
+
+    if (!events) {
+      throw new Error(`${trigger} is missing event definition(s)`);
+    }
+
+    events.forEach((event: EventMethods) => {
+      const eventListener: () => void = this.renderer.listen(
+        this.reference?.nativeElement,
+        event.event,
+        event.method
+      );
+      this.eventListeners.push(eventListener);
+    });
+  }
+
+  private handleClickInsideHostElement(): void {
+    if (this.closeOnSelect) {
+      this.hide();
+    }
+  }
+
+  private handleClickOutsideHostElement(event: Event): void {
+    const clickedOnReferenceWithClickTriggerEnabled: boolean =
+      this.reference?.nativeElement.contains(event.target) && this.triggers.includes('click');
+
+    if (this.closeOnBackdrop && !clickedOnReferenceWithClickTriggerEnabled) {
+      this.hide();
+    }
+  }
+
+  private tearDownEventHandling(): void {
+    this.eventListeners.forEach((eventListener) => {
+      if (eventListener != null) {
+        eventListener();
+      }
+    });
+    this.eventListeners = [];
```

Comment:
For performance reasons, to not risk collecting a lot of detached elements, shouldnt we actually remove any attached event-listeners here instead?

Reply:
Can you elaborate. The if statement should not be necessary, but just in-case so we are sure to be able to execute the function

Reply:
It dawns on me that I misread this, I was not aware that this utilizes the teardown function returned from renderer. 

But that hightlights the actual problem here, I believe: The naming is a bit misleading, because the value returned and pushed to the eventListeners array is not actually an event listener, but rather an `EventListenerDisposeFunction` - so my comment from before about removing the EventListener is actually wrong, it should be replaced with something like that instead. And then we should rename the array to reflect that.